src/sentry/incidents/subscription_processor.py

```diff
@@ -259,10 +260,61 @@ def get_crash_rate_alert_aggregation_value(subscription_update):
+        metrics dataset.
+        The subscription update looks like
+        [
+            {'project_id': 8, 'tags[5]': 6, 'value': 2.0},
+            {'project_id': 8, 'tags[5]': 13,'value': 1.0}
+        ]
+        where each entry represents a session status and the count of that specific session status.
+        As an example, `tags[5]` represents string `session.status`, while `tags[5]: 6` could
+        mean something like there are 2 sessions of status `crashed`. Likewise the other entry
+        represents the number of sessions started. In this method, we need to reverse match these
+        strings to end up with something that looks like
+        {"init": 2, "crashed": 4}
+        - `init` represents sessions or users sessions that were started, hence to get the crash
+        free percentage, we would need to divide number of crashed sessions by that number,
+        and subtract that value from 1. This is also used when CRASH_RATE_ALERT_MINIMUM_THRESHOLD is
+        set in the sense that if the minimum threshold is greater than the session count,
+        then the update is dropped. If the minimum threshold is not set then the total sessions
+        count is just ignored
+        - `crashed` represents the total sessions or user counts that crashed.
+        """
+        session_status = tag_key(self.subscription.project.organization.id, "session.status")
+        data = {}
+
+        for row in subscription_update["values"]["data"]:
+            tag_value = reverse_tag_value(
+                self.subscription.project.organization.id, row[session_status]
+            )
+            data[tag_value] = row["value"]
+
+        total_session_count = data.get("init", 0)
+        crash_count = data.get("crashed", 0)
+
+        if total_session_count == 0:
+            metrics.incr("incidents.alert_rules.ignore_update_no_session_data")
+            return
+
+        if CRASH_RATE_ALERT_MINIMUM_THRESHOLD is not None:
+            min_threshold = int(CRASH_RATE_ALERT_MINIMUM_THRESHOLD)
+            if total_session_count < min_threshold:
+                metrics.incr("incidents.alert_rules.ignore_update_count_lower_than_min_threshold")
+                return
+
+        aggregation_value = round((1 - crash_count / total_session_count) * 100, 3)
+        return aggregation_value
+
     def get_aggregation_value(self, subscription_update):
-        is_sessions_dataset = Dataset(self.subscription.snuba_query.dataset) == Dataset.Sessions
-        if is_sessions_dataset:
+        if self.subscription.snuba_query.dataset == QueryDatasets.SESSIONS.value:
             aggregation_value = self.get_crash_rate_alert_aggregation_value(subscription_update)
+        elif self.subscription.snuba_query.dataset == QueryDatasets.METRICS.value:
+            aggregation_value = self.get_crash_rate_alert_metrics_aggregation_value(
+                subscription_update
+            )
```

Comment:
Does this mean that any subscription update received for the metrics dataset will be treated as a session crash rate result? If yes, I guess that's fine for now, but we will potentially have many different use cases for metrics-based alerts in the future.

Reply:
don't we get some query ID back from snuba on which we can dispatch?

Reply:
Yep we get a `subscription_id` that can be used for that purpose I guess
fwiw this is what the payload returned looks like 
```
{'subscription_id': '0/b2ec54a04de411ecb2e1acde48001122', 'request': {'query': 'MATCH (metrics_counters) SELECT sum(value) AS value BY project_id, tags[3] WHERE org_id = 1 AND project_id IN array(1) AND metric_id = 7', 'parent_api': '<unknown>'}, 'result': {'data': [{'project_id': 1, 'tags[3]': 13, 'value': 8.0}, {'project_id': 1, 'tags[3]': 4, 'value': 46.0}], 'meta': [{'name': 'project_id', 'type': 'UInt64'}, {'name': 'tags[3]', 'type': 'UInt64'}, {'name': 'value', 'type': 'Float64'}]}, 'timestamp': datetime.datetime(2021, 11, 25, 13, 1, tzinfo=<UTC>), 'values': {'data': [{'project_id': 1, 'tags[3]': 13, 'value': 8.0}, {'project_id': 1, 'tags[3]': 4, 'value': 46.0}], 'meta': [{'name': 'project_id', 'type': 'UInt64'}, {'name': 'tags[3]', 'type': 'UInt64'}, {'name': 'value', 'type': 'Float64'}]}}
```

Reply:
If you want these to do different things then that could live on the `AlertRule` itself, which has a link to the subscription via the `subscription_id`